File: scripts/generate-logos.py
#!/usr/bin/env python3
"""Generate logo assets from SVG files."""

import subprocess
import sys

def install_deps():
    """Install required dependencies."""
    subprocess.check_call([sys.executable, "-m", "pip", "install", "cairosvg", "Pillow", "-q"])

try:
    import cairosvg
    from PIL import Image
    import io
except ImportError:
    print("Installing dependencies...")
    install_deps()
    import cairosvg
    from PIL import Image
    import io

from pathlib import Path

PUBLIC_DIR = Path(__file__).parent.parent / "public"

def svg_to_png(svg_path: Path, output_path: Path, size: int):
    """Convert SVG to PNG at specified size."""
    png_data = cairosvg.svg2png(
        url=str(svg_path),
        output_width=size,
        output_height=size
    )
    with open(output_path, "wb") as f:
        f.write(png_data)
    print(f"Created: {output_path.name} ({size}x{size})")

def create_favicon(icon_svg: Path, output_path: Path):
    """Create favicon.ico with multiple sizes."""
    sizes = [16, 32, 48]
    images = []

    for size in sizes:
        png_data = cairosvg.svg2png(
            url=str(icon_svg),
            output_width=size,
            output_height=size
        )
        img = Image.open(io.BytesIO(png_data))
        images.append(img)

    images[0].save(
        output_path,
        format="ICO",
        sizes=[(s, s) for s in sizes],
        append_images=images[1:]
    )
    print(f"Created: {output_path.name} (16x16, 32x32, 48x48)")

def main():
    icon_svg = PUBLIC_DIR / "luis-travel-icon.svg"
    logo_svg = PUBLIC_DIR / "luis-travel-logo.svg"

    if not icon_svg.exists():
        print(f"Error: {icon_svg} not found")
        return 1

    # Generate PWA icons from icon SVG
    svg_to_png(icon_svg, PUBLIC_DIR / "logo192.png", 192)
    svg_to_png(icon_svg, PUBLIC_DIR / "logo512.png", 512)

    # Generate favicon
    create_favicon(icon_svg, PUBLIC_DIR / "favicon.ico")

    # Generate icon PNG for inline use
    svg_to_png(icon_svg, PUBLIC_DIR / "luis-travel-icon.png", 200)

    # Generate logo PNG for header use
    if logo_svg.exists():
        png_data = cairosvg.svg2png(
            url=str(logo_svg),
            output_width=400,
            output_height=120
        )
        output_path = PUBLIC_DIR / "luis-travel-logo.png"
        with open(output_path, "wb") as f:
            f.write(png_data)
        print(f"Created: {output_path.name} (400x120)")

    print("\nAll logo assets generated successfully!")
    return 0

if __name__ == "__main__":
    sys.exit(main())

File: scripts/generate-logos.mjs
#!/usr/bin/env node
/**
 * Generate logo assets from SVG files
 */

import { readFileSync, writeFileSync } from 'fs';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';
import sharp from 'sharp';

const __dirname = dirname(fileURLToPath(import.meta.url));
const PUBLIC_DIR = join(__dirname, '..', 'public');

async function svgToPng(svgPath, outputPath, size) {
  const svg = readFileSync(svgPath);
  await sharp(svg)
    .resize(size, size)
    .png()
    .toFile(outputPath);
  console.log(`Created: ${outputPath.split('/').pop()} (${size}x${size})`);
}

async function createFavicon(svgPath, outputPath) {
  const svg = readFileSync(svgPath);
  // Create 32x32 PNG for favicon (sharp doesn't do ICO, but modern browsers accept PNG)
  await sharp(svg)
    .resize(32, 32)
    .png()
    .toFile(outputPath.replace('.ico', '.png'));

  // For ICO, we'll create a 32x32 version
  const png32 = await sharp(svg).resize(32, 32).png().toBuffer();
  const png16 = await sharp(svg).resize(16, 16).png().toBuffer();

  // Write as PNG since ICO requires special handling
  writeFileSync(outputPath.replace('.ico', '-32.png'), png32);
  writeFileSync(outputPath.replace('.ico', '-16.png'), png16);
  console.log(`Created: favicon PNGs (16x16, 32x32)`);
}

async function main() {
  const iconSvg = join(PUBLIC_DIR, 'luis-travel-icon.svg');
  const logoSvg = join(PUBLIC_DIR, 'luis-travel-logo.svg');

  try {
    // Generate PWA icons
    await svgToPng(iconSvg, join(PUBLIC_DIR, 'logo192.png'), 192);
    await svgToPng(iconSvg, join(PUBLIC_DIR, 'logo512.png'), 512);

    // Generate icon PNG for inline use
    await svgToPng(iconSvg, join(PUBLIC_DIR, 'luis-travel-icon.png'), 200);

    // Generate favicon
    await createFavicon(iconSvg, join(PUBLIC_DIR, 'favicon.ico'));

    // Generate logo PNG
    const logoSvgContent = readFileSync(logoSvg);
    await sharp(logoSvgContent)
      .resize(400, 120)
      .png()
      .toFile(join(PUBLIC_DIR, 'luis-travel-logo.png'));
    console.log(`Created: luis-travel-logo.png (400x120)`);

    console.log('\nAll logo assets generated successfully!');
  } catch (error) {
    console.error('Error generating logos:', error.message);
    process.exit(1);
  }
}

main();
